compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -1284,7 +1284,9 @@ class Definitions {
 
   private val ScalaImportFns: List[RootRef] =
     JavaImportFns :+
-    RootRef(() => ScalaPackageVal.termRef)
+    RootRef(() => ScalaPackageVal.termRef) :+
+    RootRef(() => ScalaPackageObject.termRef)
```

Comment:
This just makes compilation slower. An import from ScalaPackageVal will automatically import members of `scala.package`. On the other hand, now we have one more import to search for every root reference. 

What breaks if we do not do this change?

Reply:
Without this change the PR doesn't fix anything. But I agree that we shouldn't need it.

Reply:
As stated in the PR description, this PR fixes 2 issues (caused by 2 separate things in the code base but influencing users in a similar way):
- Completion of symbols from scala package object, like `Seq` or `BigDecimal`
- Completion of other predefined symbols (from Predef and top level definitions from scala package), like `println`
The latter issue is indeed a regression introduced in the PR that @smarter mentioned and it occurs only in LSP. The former issue must be older, even though nobody else seems to have spotted it before, it affects both LSP and REPL and is fixed specifically by this line (although this itself wouldn't fix LSP without the other fix).
I'm not really sure how the problem could be solved without adding this line. Interestingly this only changes the behaviour of symbols completion. Symbols like `Seq` or `BigDecimal` cause no problems in compilation and symbols completion should probably have exactly the same view of symbols in scope as the compiler.

Reply:
Ah right, I forgot the other issue. Adding root imports to the context like you did is indeed the right solution.